framework/src/iotbus/iotbus_gpio.c

```diff
@@ -375,93 +394,164 @@ int iotbus_gpio_unregister_cb(iotbus_gpio_context_h dev)
 	return IOTBUS_ERROR_NONE;
 }
 
-#ifdef CONFIG_IOTDEV
-int iotbus_gpio_set_interrupt(iotbus_gpio_context_h dev, iotbus_int_type_e int_type, iotbus_gpio_cb cb)
+static pthread_addr_t gpio_intr_thread(pthread_addr_t arg)
 {
-	int fd, ret, edge;
+	struct intr_args *param = (struct intr_args *)arg;
+	iotbus_gpio_context_h dev = (iotbus_gpio_context_h)param->dev;
 	struct _iotbus_gpio_s *handle;
-	struct iotbus_int_info_s info = { 0, };
+	
+	siginfo_t info;
+	sigset_t sig_set;
+	uint32_t val;
 
-	handle = (struct _iotbus_gpio_s *)dev->handle;
-	handle->cb = cb;
+	sigemptyset(&sig_set);
+	sigaddset(&sig_set, SIG_IOTBUS_GPIO_RISING);
+	sigaddset(&sig_set, SIG_IOTBUS_GPIO_FALLING);
 
-	info.handle = dev;
-	info.pin_type = IOTBUS_GPIO;
-	info.int_type = int_type;
-	info.pid = iotapi_get_pid();
-	
-	fd = open(IOTBUS_SIGPATH, O_NONBLOCK);
-	if (fd < 0) {
-		ret = get_errno();
-		ibdbg("Open iotsig fail[%d, %d]\n", fd, ret);
-		return IOTBUS_ERROR_DEVICE_FAIL;
-	}
+	pthread_sigmask(SIG_BLOCK, &sig_set, NULL);
```

Comment:
The original sigset will be saved into the 3rd param. So you can restore the original sigset from line 434.